src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -612,13 +548,13 @@ private void removeTerritoriesThatArentWorthAttacking(final List<ProTerritory> p
         int minCost = Integer.MAX_VALUE;
         Unit minUnit = null;
         for (final Unit u : t.getUnits().getMatches(Matches.unitIsOwnedBy(player))) {
-          if (playerCostMap.getInt(u.getType()) < minCost) {
-            minCost = playerCostMap.getInt(u.getType());
+          if (ProData.unitValueMap.getInt(u.getType()) < minCost) {
```

Comment:
Data classes with mutable data are an anti-pattern. 

- Put functionality into them, repeated client code, whatever so the level of abstraction for clients is greater compared to just dealing with the raw primitives which you are wrapping
- Go the opposite route and remove functionality. Make them "data messenger" classes by converting the object to be immutable. Or go further and consider replacing with a read only interface which can be implemented easily with anonymous inner classes.

IE: instead of 
```
public class MyData {
  private String thisIsAnAntiPattern = null;
  public MyData() {  }
  public void setAntiPattern(String statefulData) { this.thisIsAnAntiPattnern = statefulData; }
  public String getThisIsAnAntiPattern() { return thisIsAnAntiPattern;  }
}

MyData data= new MyData();
data.setThisIsAnAntiPattern("my stateful data which I'm tracking around");
foobar( data );
```

Consider:
```
public interface MyData {
    public String slightlyBetter();
}

final String myData = "some data"
foobar( new MyData() { @Override public String slightlyBetter() { return myData } }
```

Then if you have the luxury of only need one data item, then you can make it a functional interface and simplify it a lot more:
```
@FunctionalInterface 
public interface MyData {
    public String slightlyBetter();
}

final String myData = "some data"
foobar( () -> myData );
```